pkg/cache/cache.go

```diff
@@ -109,10 +108,9 @@ func (m *MemcachedCache) Get(k string) ([]byte, error) {
 	return item.Value, nil
 }
 
-func (m *MemcachedCache) Set(k string, v []byte, expire int32) {
-	key := sha256.Sum256([]byte(k))
-	hk := hex.EncodeToString(key[:])
-	go m.client.Set(&memcache.Item{Key: m.prefix + hk, Value: v, Expiration: expire})
+func (m *MemcachedCache) Set(k string, v []byte, expire int32) error {
+	hk := getCacheHashKey(k)
+	return m.client.Set(&memcache.Item{Key: m.prefix + hk, Value: v, Expiration: expire})
```

Comment:
Set will be sync after that change, right? I know that's easier for error but will set timeout work in that case?

Reply:
Set function of other caches are sync. It made no sense to have an inconsistency over the types implementing the same interface. So, I made this consistent with others (sync).

Reply:
Ok, makes sense then.